wgpu-core/src/instance.rs

```diff
@@ -118,6 +118,46 @@ impl Instance {
         }
     }
 
+    pub fn from_hal<A: HalApi>(name: &str, raw_instance: A::Instance) -> Self {
+        #[cfg(vulkan)]
+        let mut vulkan_instance = None;
```

Comment:
can probably start with `let mut this = Self::default()` and mutate it, to save a ton of lines

Reply:
I'm not sure how I would use this. The problem I was having here, is that `<A as hal::api::Api>::Instance` cannot be converted to `hal::api::Vulkan::Instance`. I'm confused since `wgc::Instance::new()` seems so be doing just that.

Reply:
What I'm thinking is:
```rust
let mut this = Self::default();
this.name = name.to_string();
match A::VARIANT {
...
}
Ok(this)
```

Reply:
it seems that the problem is that matching by `VARIANT` doesn't actually constrain the type `A`. Maybe this could be fixed with full const generics but we don't have that yet. For now I can wrap hal::Instance into an enum.

Reply:
right, we can't have this method as generic over A

Reply:
hmm, or maybe we can. There is `get_surface_mut`, which works with generic A. Something similar can be done for instances

Reply:
I tried a few more type manipulations but still no luck. The enum solution works.

Reply:
I did it, at least I think I did what you suggested. I used the `HalApi` trait to convert from hal to wgc instance (while `get_surface_mut` converts a wgc swapchain to hal).